feat(documentation) Adds the november 2022 newsletter (#5760)
jenkins-infra/jenkins.io: content/blog/2022/12/2022-12-05-jenkins-newsletter.adoc

---
layout: post
title: "Jenkins November 2022 Newsletter"
tags:
- jenkins
- newsletter
- community
- contribute
authors:
- alyssat
opengraph:
  image: /images/post-images/2022-10-13-jenkins-newsletter/centered-newsletter.png
links:
  discourse: true
---

== November 2022

> image:/images/post-images/2022-10-13-jenkins-newsletter/centered-newsletter.png[Jenkins September Newsletter]

Reviewer: ```suggestion
image:/images/post-images/2022-10-13-jenkins-newsletter/centered-newsletter.png[Jenkins November 2022 Newsletter]
```

Welcome to the Jenkins Newsletter! This is a compilation of progress within the project, highlighted by Jenkins Special Interest Groups (SIGs) for the month of November.

Got Inspiration? We would love to highlight your cool Jenkins innovations.
Share https://docs.google.com/forms/d/e/1FAIpQLScMCGOMtn2hGpfXsbyssGhVW1LwlW4LkXCIaKINKDQU2m6ieg/viewform[your story] and you could be in the next Jenkins newsletter.
View previous editions of the Jenkins Newsletter link:/node/tags/newsletter/[*here*]!

Happy reading!

== Highlights:

* Congratulations to the new officers and members of the governance
board.
* Pipeline Utility Steps Plugin was impacted by
https://commons.apache.org/proper/commons-configuration/security.html#CVE-2022-33980_prior_to_2.8.0.2C_RCE_when_applied_to_untrusted_input[CVE-2022-33980],
leading to a remote code execution in Jenkins.

> * New “jenkinsci” NPM official account for Jenkins project

Reviewer 2: ```suggestion
* New “jenkinsci” NPM official account for Jenkins project.
```

> * New “jenkinsci” NPM official account for Jenkins project.

Reviewer: ```suggestion
* New "jenkinsci" NPM official account for Jenkins project.
```

> * ssh-agent docker image is now using a https://github.com/jenkinsci/docker-ssh-agent/commit/9abba2cf2b286ad35bdb0df634c3e471da373478[correct volume] for agent workspace

Reviewer 2: ```suggestion
* ssh-agent docker image is now using a https://github.com/jenkinsci/docker-ssh-agent/commit/9abba2cf2b286ad35bdb0df634c3e471da373478[correct volume] for agent workspace.
```

* Jenkins 2.375.1 was released with more improvements to the user interface.

> * Jenkins.io is now utilizing Algolia v3 for its search feature.

Reviewer: ```suggestion
* Jenkins.io is now using Algolia v3 for its search feature.
```
(nit)

* Jenkins in Google Summer of Code 2023, preparations to help new contributors prepare effective proposals.

== Governance Update

Contributed by: https://github.com/MarkEWaite[Mark Waite]

The 2022 Jenkins elections are complete.
Thanks to Kevin Martens for the https://www.jenkins.io/blog/2022/11/17/jenkins-election-candidates-and-results/[blog post] announcing the results of the election and thanks to Damien Duportal for running the election process.
We are glad to have received 65 registrations to vote in the election.
That is less than the 81 registered voters we had in 2021.

Elected officers and members of the Jenkins governance board will begin their service December 3, 2022.
Special thanks to the members of the governance board that have completed their two-year service:

* https://www.jenkins.io/blog/authors/halkeye/[Gavin Mogan]
* https://www.jenkins.io/blog/2021/03/22/governance-updates/#about-ewelina-wilkosz[Ewelina Wilkosz]

Congratulations to the new members of the governance board.

* https://www.jenkins.io/blog/authors/uhafner/[Ullrich Hafner]
* https://www.jenkins.io/blog/authors/notmyfault/[Alexander Brandes]

Congratulations also to the newly elected officers:

> * Release Officer - https://www.jenkins.io/blog/authors/timja/[Tim Jacomb]

Reviewer 3: shouldn't this be like `link:/blog/authors/timja/` or something that mark always suggests?

Reviewer 4: I think it should be this:
```suggestion
* Release Officer - author:timja[]
```

* Events Officer - https://www.jenkins.io/blog/authors/alyssat/[Alyssa Tong]
* Security Officer - https://www.jenkins.io/blog/authors/wadeck/[Wadeck Follonier]
* Documentation Officer - https://www.jenkins.io/blog/authors/kmartens27/[Kevin Martens]
* Infrastructure Officer - https://www.jenkins.io/blog/authors/dduportal/[Damien Duportal]

Governance board meetings are held every two weeks. Meeting notes and
recordings of the meetings are available on the
https://community.jenkins.io/tag/governance[_Jenkins community site_].

== Security Update

Contributed by: https://github.com/Wadeck[Wadeck Follonier]

* One security advisory during November about plugins
** Usually the vulnerabilities in third party dependencies are not impacting the final product, it is worth mentioning when there is a true
positive case.
Pipeline Utility Steps Plugin was impacted by https://commons.apache.org/proper/commons-configuration/security.html#CVE-2022-33980_prior_to_2.8.0.2C_RCE_when_applied_to_untrusted_input[CVE-2022-33980], leading to a remote code execution in Jenkins.

== Infrastructure Update

Contributed by: https://github.com/dduportal[Damien Duportal]

> * Upgrade of our controllers to the latest LTS 2.375.1.

Reviewer: ```suggestion
* Upgrade of our controllers to the latest LTS 2.375.1
```
I find this dot confusing (just mentioning, feel free to keep it)

Author: What about using that notation `2.375.1`. ?

Reviewer 5: +1 with @gounthar 

- Sentences need a dot. Wether we find the rendering confusing or not.
- Using "inline code" rendering breaks the visual rendering so that might help removing confusion

```suggestion
* Upgrade of our controllers to the latest LTS `2.375.1`.
```

> * Upgrade of our controllers to the latest LTS 2.375.1

Reviewer 2: ```suggestion
* Upgrade of our controllers to the latest LTS 2.375.1.
```

Reviewer: I find this last dot confusing on this version number.

> * New “jenkinsci” NPM official account for Jenkins project

Reviewer 2: ```suggestion
* New “jenkinsci” NPM official account for Jenkins project.
```

> * New “jenkinsci” NPM official account for Jenkins project.

Reviewer: ```suggestion
* New "jenkinsci" NPM official account for Jenkins project.
```

> * JDK17 for Windows provided to developers

Reviewer 2: ```suggestion
* JDK17 for Windows provided to developers.
```

> * https://meetings.jenkins-ci.org/[https://meetings.jenkins-ci.org/] website (archive Board meeting minutes) is recovered and back

Reviewer 2: ```suggestion
* https://meetings.jenkins-ci.org/[https://meetings.jenkins-ci.org/] website (archive Board meeting minutes) is recovered and back.
```

> * Deprecated plugin “windows-slaves” removed

Reviewer 2: ```suggestion
* Deprecated plugin “windows-slaves” removed.
```

> * Deprecated plugin “windows-slaves” removed.

Reviewer: ```suggestion
* Deprecated "windows-slaves" plugin removed.
```

* Azure networks are now code-managed to allow incoming IPv6 support and

> improved network performances

Reviewer 2: ```suggestion
improved network performances.
```

Author: Thank you so much for your review @alyssat 🙏

== Platform Update

Contributed by: https://github.com/gounthar[Bruno Verachten]

* Docker Images for Jenkins changes:

> ** JDK17: added https://github.com/jenkinsci/docker-agent/commit/0217d06c19e989737d5dd3ed7e0116e628c2bf2f[ARMv7 support]

Reviewer 2: ```suggestion
** JDK17: added https://github.com/jenkinsci/docker-agent/commit/0217d06c19e989737d5dd3ed7e0116e628c2bf2f[ARMv7 support].
```

> ** Monthly Base OS https://github.com/jenkinsci/docker-agent/commit/a1ca8ef2224f46cd688abd71d624764adc3606e8[updates] and autumn cleanup

Reviewer 2: ```suggestion
** Monthly Base OS https://github.com/jenkinsci/docker-agent/commit/a1ca8ef2224f46cd688abd71d624764adc3606e8[updates] and autumn cleanup.
```

> ** Huge work (not finished yet) to ensure “latest” version https://github.com/jenkinsci/docker-inbound-agent/issues/271[being overridden] by older releases

Reviewer 2: ```suggestion
** Huge work (not finished yet) to ensure “latest” version https://github.com/jenkinsci/docker-inbound-agent/issues/271[being overridden] by older releases.
```

> ** https://github.com/jenkinsci/docker-agent/commit/a5272d089460267362a2ccebb3f9b36931adb16d[Automatic tracking] of remoting and JDK versions implemented: we will get faster updates

Reviewer 2: ```suggestion
** https://github.com/jenkinsci/docker-agent/commit/a5272d089460267362a2ccebb3f9b36931adb16d[Automatic tracking] of remoting and JDK versions implemented: we will get faster updates.
```

> ** A LOT of fixes (missing packages such as https://github.com/jenkinsci/docker-agent/commit/70c7dba8237b2e7c808c88bb007efa771904d6fd[curl], https://github.com/jenkinsci/docker-agent/commit/e8a6599dff3f38e45d720d32dfb2d2d1f07662af[ssh, git], https://github.com/jenkinsci/docker-ssh-agent/commit/acde0aa06e3de1c95e02f5a742c912cc02654042[less, patch])

Reviewer 2: ```suggestion
** A LOT of fixes (missing packages such as https://github.com/jenkinsci/docker-agent/commit/70c7dba8237b2e7c808c88bb007efa771904d6fd[curl], https://github.com/jenkinsci/docker-agent/commit/e8a6599dff3f38e45d720d32dfb2d2d1f07662af[ssh, git], https://github.com/jenkinsci/docker-ssh-agent/commit/acde0aa06e3de1c95e02f5a742c912cc02654042[less, patch]).
```

> ** ssh-agent is now using https://github.com/jenkinsci/docker-ssh-agent/commit/9abba2cf2b286ad35bdb0df634c3e471da373478[correct volume] for agent workspace

Reviewer 2: ```suggestion
** ssh-agent is now using https://github.com/jenkinsci/docker-ssh-agent/commit/9abba2cf2b286ad35bdb0df634c3e471da373478[correct volume] for agent workspace.
```

* New platforms:

> ** Experiments with Windows 2022: working with Jenkins Infrastructure to build these images

Reviewer 2: ```suggestion
** Experiments with Windows 2022: working with Jenkins Infrastructure to build these images.
```

> * Experiments with JDK19: ANTLR 2 to ANTLR 4 transition complete, Jenkins core compiles

Reviewer 2: ```suggestion
* Experiments with JDK19: ANTLR 2 to ANTLR 4 transition complete, Jenkins core compiles.
```

> == User Experience Update [Mark Waite]

Reviewer 2: ```suggestion
== User Experience Update 
```

Contributed by: https://github.com/MarkEWaite[Mark Waite]

Jenkins 2.375.1 was released on November 30, 2022 with more improvements to the user interface.
Special thanks to Jan Faracik, Tim Jacomb, Alexander Brandes, Wadeck Follonier, Daniel Beck, and many others that were involved in the most recent improvements to the user experience.

The weather icons have been updated to be more recognizable. They continue to work well in both light themes and dark themes.

image:/images/post-images/2022-12-05-jenkins-newsletter/10000000000003C00000021C2C1F8B51560D56F4.png[image,width=624,height=350]

> The plugin manager navigation has moved from the top of the page to the side panel. The search field is more visible

Reviewer 2: ```suggestion
The plugin manager navigation has moved from the top of the page to the side panel. The search field is more visible.
```

image:/images/post-images/2022-12-05-jenkins-newsletter/10000000000003C00000021C7524A94118BE3493.png[image,width=624,height=350]

The User Experience SIG is also pleased to note a valuable improvement in the https://www.jenkins.io/changelog/#v2.380[Jenkins 2.380 weekly release].
The tooltips that were previously provided by the unmaintained and long outdated YahooUI JavaScript library are now being provided by the Tippy.js JavaScript framework.
Special thanks to Jan Faracik for his work removing that use of the YahooUI JavaScript library.

> == Documentation Update [Kevin Martens]

Reviewer 2: ```suggestion
== Documentation Update 
```

Contributed by: https://github.com/kmartens27[Kevin Martens]

> Jenkins.io is now utilizing Algolia v3 for its search feature.

Reviewer: ```suggestion
Jenkins.io is now using Algolia v3 for its search feature.
```
(nit)

This update has not only improved searching on the Jenkins site, but also provided a new search UI, which provides helpful suggestions.
Massive thanks to Gavin Mogan for working on this and improving the Jenkins.io search.

image:/images/post-images/2022-12-05-jenkins-newsletter/1000020100000240000000403C0CC85DFBCE3E67.png[image,width=576,height=63]

image:/images/post-images/2022-12-05-jenkins-newsletter/10000201000004B2000002B55DF2DC7EFDD5C5DB.png[image,width=624,height=360]

> Algolia has graciously upgraded our search from their legacy v2 to the super pretty and useful v3 apis.
This includes a new fully accessible popup. I just love being able to goto jenkins.io and hitting ctrl+k to
search.

said https://www.jenkins.io/blog/authors/halkeye/[Gavin Mogan], current Jenkins Board Member, maintainer of the Jenkins plugin site, and plugin site API.

== Advocacy & Outreach Update

Contributed by: https://github.com/alyssat[Alyssa Tong]

Jenkins gets ready for Google Summer of Code 2023!

Google recently announced the GSoC https://developers.google.com/open-source/gsoc/timeline[2023 program timeline], and the Advocacy & Outreach SIG has responded!

> We’ve established the https://www.jenkins.io/blog/2022/11/23/get-prepared-for-gsoc/[GSoC early preparations for applicants - steps to effective submission] post to help future contributors with the process.

Reviewer: ```suggestion
We've established the https://www.jenkins.io/blog/2022/11/23/get-prepared-for-gsoc/[GSoC early preparations for applicants - steps to effective submission] post to help future contributors with the process.
```

On December 20, 2022 at 4PM UTC there will be a walk through of this process via a webinar.
We would like this to be an interactive webinar so bring your questions.
See https://www.jenkins.io/events/[Event Calendar] (See GSoC 2023 - Contributor webinar: How to get ready) for login details.

We are still in great need of https://www.jenkins.io/blog/2022/11/16/gsoc-2023/[project idea proposals and mentors].

* GSoC project ideas are coding projects that potential GSoC contributors can accomplish in 10-22 weeks.
* The coding projects can be new features, plugins, test frameworks, infrastructure, etc. Anyone can submit a project idea.
* Mentoring takes about 5 to 8 hours of work per week (more at the start, less at the end).

> * Mentors provide guidance, coaching, and sometimes a bit of cheerleading. They review GSoC contributor proposals, pull-requests and contributor presentations at the evaluation phase.

Reviewer: ```suggestion
* Mentors provide guidance, coaching, and sometimes a bit of cheerleading. They review GSoC contributor proposals, pull requests and contributor presentations at the evaluation phase.
```

* JDK17 for Windows provided to developers.
* https://meetings.jenkins-ci.org/[https://meetings.jenkins-ci.org/] website (archive Board meeting minutes) is recovered and back.
* Deprecated "windows-slaves" plugin removed.

> * Azure networks are now code-managed to allow incoming IPv6 support and improved network performances.

Reviewer 5: That is work in progress and both the user visible results (ipv6 and improved network performance) are not there yet 😅
```suggestion
```
